channel/appregistry_internal_test.go

```diff
@@ -102,13 +102,50 @@ func assertIdentity(t *testing.T, expected App) {
 }
 
 func newRandomMockApp(rng *rand.Rand) App {
-	return NewMockApp(wtest.NewRandomAddress(rng))
+	return NewMockApp(newRandomAppID(rng))
 }
 
 func resetAppRegistry() {
 	appRegistry.Lock()
 	defer appRegistry.Unlock()
 	appRegistry.resolvers = nil
-	appRegistry.singles = make(map[wallet.AddrKey]App)
+	appRegistry.singles = make(map[AppIDKey]App)
 	appRegistry.defaultRes = nil
 }
+
+func newRandomAppID(rng *rand.Rand) AppID {
+	id := appID{}
+	rng.Read(id[:])
+	return id
+}
+
+const appIDLength = 32
+
+type appID [appIDLength]byte
+
+func (id appID) MarshalBinary() (data []byte, err error) {
+	return id[:], nil
+}
+
+func (id appID) UnmarshalBinary(data []byte) error {
+	l := len(data)
+	if l != appIDLength {
+		return fmt.Errorf("invalid length: %v", l)
+	}
+	copy(id[:], data)
+	return nil
+}
+
+func (id appID) Equal(b AppID) bool {
+	bTyped, ok := b.(appID)
+	return ok && bytes.Equal(id[:], bTyped[:])
+}
+
+// Key returns the object key which can be used as a map key.
```

Comment:
You could remove this doc since the other functions also have no doc and its explained in the interface imo.

Reply:
Removed